src/sql-parser/src/parser.rs

```diff
@@ -1825,6 +1826,74 @@ impl<'a> Parser<'a> {
         };
         Ok(compression)
     }
+    /// Parse set of table definitions out of literal string from within CREATE SOURCE
+    /// 'tbl1_oid AS tbl1_name (col1_name col1_type, col2_name col2_type), tbl2_oid AS tbl2_name (col1_name col1_type,...)'
+    fn parse_source_tables(&mut self) -> Result<Vec<PgTable<Raw>>, ParserError> {
+        let mut tables: Vec<PgTable<Raw>> = Vec::new();
+        let raw_def = self.parse_literal_string()?;
+        if raw_def.is_empty() {
+            return Ok(tables);
+        }
+        let tokens = lexer::lex(&raw_def)?;
+        let mut parser = Parser::new(&raw_def, tokens);
```

Comment:
I'm just dropping in without full context, but this looks very sketchy to me. Why is this syntax getting nested inside of a literal string? Based on a quick scan it doesn't seem like the AstDisplay path escapes the contents of the string, either?

Reply:
This was following the convention already in place for the Postgres source where PUBLICATION and SLOT were blocks of options in a single quoted string using the [escaping function in AstDisplay](https://github.com/nharring-adjacent/materialize/blob/main/src/sql-parser/src/ast/display.rs#L170-L180) which may or may not be sufficient for this use case. In all of the existing test cases including tables with literal spaces in their names it has proved robust though.